src/js/containers/agencyLanding/AgencyLandingContainer.jsx

```diff
@@ -0,0 +1,245 @@
+        }
+        if (this.props.autocompleteAgencies !== prevProps.autocompleteAgencies) {
+            // search input changed
+            this.fetchAgencies();
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.agenciesRequest) {
+            this.agenciesRequest.cancel();
+        }
+    }
+
+    setAgencySearchString(agencySearchString) {
+        this.setState({
+            agencySearchString
+        });
+    }
+
+    setNoResults(noResults) {
+        this.setState({
+            noResults
+        });
+    }
+
+    showColumns() {
+        const columns = [];
+        const sortOrder = AgenciesTableFields.defaultSortDirection;
+        const widths = AgenciesTableFields.columnWidthPercentage;
+
+        AgenciesTableFields.order.forEach((col) => {
+            let displayName = AgenciesTableFields[col];
+            if ((col === 'budget_authority_amount') ||
+                (col === 'percentage_of_total_budget_authority')) {
+                // Add (FY YYYY) to Budget Authority and Percent of Total U.S. Budget column headers
+                if (this.state.fy) {
+                    displayName = `${displayName} (FY ${this.state.currentFY})`;
+                }
+            }
+            const column = {
+                columnName: col,
+                displayName,
+                width: widths[col],
+                defaultDirection: sortOrder[col]
+            };
+            columns.push(column);
+        });
+
+        this.setState({
+            columns
+        }, () => {
+            this.fetchAgencies();
+        });
+    }
+
+    fetchAgencies() {
+        if (this.agenciesRequest) {
+            // a request is in-flight, cancel it
+            this.agenciesRequest.cancel();
+        }
+
+        this.setState({
+            inFlight: true
+        });
+
+        // generate the params
+        const params = {
+            sort: this.props.agenciesOrder.field,
+            order: this.props.agenciesOrder.direction
+        };
+
+        this.agenciesRequest = AgencyLandingHelper.fetchAllAgencies(params);
+
+        this.agenciesRequest.promise
+            .then((res) => {
+                this.setState({
+                    inFlight: false
+                });
+
+                this.parseAgencies(res.data);
+            })
+            .catch((err) => {
+                this.agenciesRequest = null;
+                if (!isCancel(err)) {
+                    this.setState({
+                        inFlight: false
+                    });
+                    console.log(err);
+                }
+            });
+    }
+
+    parseAgencies(data) {
+        const agencies = [];
+        const showAllAgencies = this.props.autocompleteAgencies.length === 0 && !this.state.noResults;
+
+        data.results.forEach((item) => {
+            // If there is no search term, show all agencies. Otherwise, only show agencies that match
+            // the search input
+            if (showAllAgencies ||
```

Comment:
Right now, the highlighted text persists even after the user has cleared the search box. I think you need to clear the `agencySearchString` value from state when the search field is cleared.
<img width="1198" alt="screen shot 2017-07-31 at 2 27 56 pm" src="https://user-images.githubusercontent.com/3594363/28792169-b0adfb18-75fc-11e7-8835-2eca90f5dd0e.png">